### PR TITLE
Added definitions of scripts and plugins.

### PR DESCRIPTION
**Pull request by ubaldot**
This is a proposal to enhance discussions around Vim scripting. 
Such proposal attempts at defining **_scripts_** and **_plugins_**.

### CODE REVIEW

File: runtime/doc/usr_41.txt [hunk 1]
```diff
@@ -1881,7 +1897,7 @@ mapping.  For these commands, any character until the end-of-line or '|' is
 included.  As a consequence of this behavior, you don't always see that
 trailing whitespace is included: >
 
-	map <F4> o#include  
+	map <F4> o#include
```
Comment (zeertzjq) [inline runtime/doc/usr_41.txt]: The trailing whitespace here is required.

Comment (ubaldot) [reply runtime/doc/usr_41.txt]: Auch! I have an autocommand that trim the whitespaces on save.

File: runtime/doc/usr_41.txt [hunk 2]
```diff
@@ -1881,7 +1897,7 @@ mapping.  For these commands, any character until the end-of-line or '|' is
...
```
Comment (zeertzjq) [inline runtime/doc/usr_41.txt]: Wrong number of trailing spaces. There should be two.

**Comment by chrisbra** [comment]
thanks!

**Comment by benknoble** [comment]
It would probably be a good idea to include something (here or elsewhere) to disambiguate from `:help plugin`, which says

```
Vim's functionality can be extended by adding plugins.  A plugin is nothing
more than a Vim script file that is loaded automatically when Vim starts.  You
can add a plugin very easily by dropping it in your plugin directory.
{not available when Vim was compiled without the |+eval| feature}

There are two types of plugins:

    global plugin: Used for all kinds of files
  filetype plugin: Only used for a specific type of file

```

**Comment by ubaldot** [comment]
Good catch! I'll try fix it in the incoming days.

**Pull request closed without merging**